primedev/engine/gl_matsysiface.cpp

```diff
@@ -0,0 +1,50 @@
+#include "materialsystem/cmaterialglue.h"
+
+CMaterialGlue* (*GetMaterialAtCrossHair)();
+
+AUTOHOOK_INIT()
+
+AUTOHOOK(CC_mat_crosshair_printmaterial_f, engine.dll + 0xB3C40, void, __fastcall, (const CCommand& args))
+{
+	CMaterialGlue* pMat = GetMaterialAtCrossHair();
+
+	if (!pMat)
+	{
+		spdlog::error("Not looking at a material!\n");
+		return;
+	}
+
+	std::function<void(CMaterialGlue * pGlue, const char* szName)> fnPrintGlue = [](CMaterialGlue* pGlue, const char* szName)
+	{
+		spdlog::info("|-----------------------------------------------\n");
+
+		if (!pGlue)
+		{
+			spdlog::info("|-- %s is NULL\n", szName);
+			return;
+		}
+
+		spdlog::info("|-- Name: %s\n", szName);
+		spdlog::info("|-- GUID: %llX\n", pGlue->m_GUID);
+		spdlog::info("|-- Name: %s\n", pGlue->m_pszName);
+		spdlog::info("|-- Width : %d\n", pGlue->m_iWidth);
+		spdlog::info("|-- Height: %d\n", pGlue->m_iHeight);
+	};
+
+	spdlog::info("|- GUID: %llX\n", pMat->m_GUID);
+	spdlog::info("|- Name: %s\n", pMat->m_pszName);
+	spdlog::info("|- Width : %d\n", pMat->m_iWidth);
+	spdlog::info("|- Height: %d\n", pMat->m_iHeight);
```

Comment:
Uh spdlog might be already adding the newline on its own :P

```log
[2024-08-20] [21:48:38] [NORTHSTAR] [info] |- GUID: %llX

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |- Width : %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |- Height: %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-----------------------------------------------

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- GUID: %llX

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Width : %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Height: %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-----------------------------------------------

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- GUID: %llX

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Width : %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Height: %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-----------------------------------------------

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- GUID: %llX

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Width : %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Height: %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-----------------------------------------------

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- GUID: %llX

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Name: %s

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Width : %d

[2024-08-20] [21:48:38] [NORTHSTAR] [info] |-- Height: %d
```

![image](https://github.com/user-attachments/assets/9890da06-897b-4d0e-ad5c-5070b74b98e1)